src/core/execute.c

```diff
@@ -5484,6 +5484,22 @@ int exec_context_destroy_credentials(const ExecContext *c, const char *runtime_p
         return 0;
 }
 
+int exec_context_destroy_mount_ns_dir(_unused_ const ExecContext *c, const char *unit) {
+        _cleanup_free_ char *p = NULL;
+
+        if (!unit)
+                return 0;
+
+        p = path_join("/run/systemd/propagate/", unit);
+        if (!p)
+                return -ENOMEM;
+
+        /* This is only filled transiently (see mount_in_namespace()), should be empty or even non-existent*/
+        (void) rmdir(p);
```

Comment:
maybe do:

```c
if (rmdir(p) < 0 && errno != ENOENT)
        log_debug_errno(errno, "Unable to remove propagation dir '%s', ignoring: %m", p);
```

i.e. usually it's a better idea to log about unexpected errors before ignoring them.

Reply:
I strove not to divert from the surrounding code that silences removal failures. (Keeping a small conservative fix, leaving broader refactoring for another opportunity.)

Reply:
debug logging is still "silent".

we changed our mindset a bit on this. previously we just (void)ed a lot of failures. These days we prefer if debug logging is added for unexpected errors. It just makes things easier to fix.

Reply:
Logging it is.